sense platform/CSUploader.m

```diff
@@ -62,35 +80,74 @@ - (BOOL) upload {
     return succeed;
 }
 
+/** Upload data batch to remote. The number of datapoints per batch is limited to ROW_LIMIT (specified at the beginning of this file).
+ 
+ @param remoteSensorIDs Dictionary containing a recent set of sensor IDs currently available on the server. Function will try to match these sensor IDs to the local sensors. If a match cannot be made it will create a new sensor. 
+ 
+ @return whether the upload was successfull or not
+ */
+- (BOOL) singleUploadWithRemoteSensorIDs: (NSDictionary *) remoteSensorIDs {
+	//get enough data from storage for a single upload
+	//TODO: check size?
+	NSArray* data = [storage getSensorDataPointsFromId:lastUploadedRowId+1 limit:ROW_LIMIT];
+	if (data.count == 0)
+		return YES;
+	//extract sensors
+	NSSet* sensors = getSensorSet(data);
+	
+	if (remoteSensorIDs == nil || remoteSensorIDs.count < sensors.count) {
+		return [self singleUpload];
```

Comment:
@jorishjanssen any way we might get here and not have upload() already resolve all sensors? Otherwise I dont think this situation is possible...